pkg/virt-launcher/virtwrap/converter/vcpu/vcpu.go

```diff
@@ -481,6 +481,9 @@ func AdjustDomainForTopologyAndCPUSet(domain *api.Domain, vmi *v12.VirtualMachin
 			log.Log.Reason(err).Error("failed to calculate passed through NUMA topology.")
 			return err
 		}
+	} else if err := numaTune(vmi, &domain.Spec, topology); err != nil {
+		log.Log.Reason(err).Error("failed to set numatune for VMI")
+		return err
```

Comment:
Can you please explain why do we need to do perform logic only if `!isNumaPassthrough(vmi)`?

Reply:
I have modified `validating` webhook not to allow specifying both `GuestMappingPassthrough` and `memoryMode`, so it made sense to me to assume that if `!isNumaPassthrough(vmi)` then it means that `memoryMode` is specified. But now I think it would be more readeable to do it like this:
```golang
      if isNumaPassthrough(vmi) {
		if err := numaMapping(vmi, &domain.Spec, topology); err != nil {
			log.Log.Reason(err).Error("failed to calculate passed through NUMA topology.")
			return err
		}
	}
	if isMemoryMode(vmi) {
	        if err := numaTune(vmi, &domain.Spec, topology); err != nil {
		         log.Log.Reason(err).Error("failed to set numatune for VMI")
		        return err
	        }
	 }
```

Reply:
@iholder101 should we rely on `validating` webhooks being triggered for every created VMI?

Reply:
Yeah, this makes a lot of sense, thanks :)
No need to separate it into two `if`s, the `else if` structure you already have is perfectly fine :+1: 
And yes, we can rely on a validating webhooks running against every VMI.